packages/wallet/api/src/sharingUnserializer.js

```diff
@@ -0,0 +1,81 @@
+// @ts-check
+import { Far, makeMarshal } from '@endo/marshal';
+
+/**
+ * Make a pair of unserializers that share low-privilege values.
+ *
+ * The "high" privilege unserializer can understand objects as slots, such as
+ * those from the wallet serializer. The "low" privilege unserializer can only
+ * understand strings as slots, such as those from the board serializer.
+ *
+ * This provides a way, for instance, to get identical brand references from
+ * both a public contract and a user's purses.
+ *
+ * @param {string} [lowProperty] - The key name to check on a high privilege
+ * slot object. If the string value at this key is equivalent to the value of
+ * another low privilege slot, the high privilege unserializer will return the
+ * same object for both slots.
+ */
+export const makeSharingUnserializer = lowProperty => {
+  assert(lowProperty, 'lowProperty must be specified for sharing unserializer');
+  const seen = {
+    // high: new WeakMap(),  // ISSUE: WeakMap conflicts with string keys
+    high: new Map(),
+    low: new Map(),
+  };
+
+  const slotToVal = {
+    /**
+     * @param {Record<string, any>} slot
+     * @param {string} iface
+     */
+    high: (slot, iface) => {
+      let obj;
+      let key;
+
+      if (lowProperty in slot) {
+        key = slot[lowProperty];
+        if (seen.low.has(key)) {
+          return seen.low.get(key);
+        }
+        obj = Far(iface, {});
+        seen.low.set(key, obj);
+      } else {
+        key = JSON.stringify(slot);
```

Comment:
`JSON.stringify` is sensitive to the insertion order of the keys, so if a caller sends `{ foo, bar }` and then later refers to the same object with `{ bar, foo }`, this will incorrectly return a new `Far` presence, even though those should be semantically identical. (I've used `djson` in Swingset for deterministic serialization, it basically just sorts the keys alphabetically).